### ISSUE TITLE
[Exporter.Geneva] Add DI support for GenevaLogExporter

### ISSUE THREAD

**Issue by shenglol** [open-telemetry/opentelemetry-dotnet-contrib#613]
# Issue with OpenTelemetry.Exporter.Geneva

List of [all OpenTelemetry NuGet packages](https://www.nuget.org/profiles/OpenTelemetry) and version that you are using:

* [OpenTelemetry.Exporter.Geneva 1.4.0-beta.1](https://www.nuget.org/packages/OpenTelemetry.Exporter.Geneva/1.4.0-beta.1)

Runtime version:

* net6.0

**Is this a feature request or a bug?**

* [x] Feature Request
* [ ] Bug

**What is the expected behavior?**

It would be great if we can add DI support for GenevaLogExporter, similar to what we have for GenevaMetricExporter and GenevaTraceExporter. This is possible now that https://github.com/open-telemetry/opentelemetry-dotnet/pull/3504 is released.

**Comment by rajkumar-rangaraj** [comment]
Could you please let us which method you want to see on Geneva Log Exporter in DI?

**Comment by shenglol** [comment]
Yes, I'm talking about [OpenTelemetryLoggerOptions.AddGenevaLogExporter](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/4a23a7e3154ab9990a0624a665e229fef0a294ac/src/OpenTelemetry.Exporter.Geneva/GenevaLoggingExtensions.cs#L31) specifically. Within the extension method `GenevaExporterOptions` is currently created via a constructor, but I think now we can use `OpenTelemetryLoggerOptions.ConfigureProvider` to get it from the DI container. What I would like to achieve with this is to bind `GenevaExporterOptions` to an `IConfiguration` outside of `AddGenevaLogExporter`.

**Comment by Gnol-VN** [comment]
Hey, I wonder if is this available yet? I would like to have this:
> configure.AddOpenTelemetry(options => options.AddProcessor<Custom_Log_Scrubber_Rule_List>());

, where I can add custom scrubbing rules through Host builder @CodeBlanch

**Comment by CodeBlanch** [comment]
Update...

* For help configuring `GenevaExporterOptions` using services retrieved through the DI container see: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Exporter.Geneva/README.md#logging

* For the question about injecting a processor using DI...

   A [new API](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4916) was released in the 1.7.0 OpenTelemetry SDK which makes the above slightly easier. You can now do this:

   ```csharp
   appBuilder.Services.AddSingleton<MyCustomProcessor>();
   appBuilder.Logging.AddOpenTelemetry(options => options.AddProcessor(
      (sp, processor) => sp.GetRequiredService<MyCustomProcessor>());
   });
   ```

**Comment by CodeBlanch** [comment]
Update...

`LoggerProviderBuilder` APIs were shipped stable in the 1.9.0 OTel SDK and support the same surface area w.r.t. DI as `TracerProviderBuilder` and `MeterProviderBuilder`. New extensions targeting `LoggerProviderBuilder` were added for GenevaExporter in its 1.9.0 release.

Logging code can now be made to work exactly like tracing & metrics.

* Using OpenTelemetry.Extensions.Hosting

```csharp
services
    .AddOpenTelemetry()
    .WithLogging(
        builder =>
        {
            builder.AddProcessor<MyProcessor>(); // Example of calling a DI-friendly API
            builder.AddGenevaLogExporter(o => o.ConnectionString = "<fill out>");
        },
        options => options.IncludeFormattedMessage = true);
```

* Using LoggerFactory.Create

```csharp
using var loggerFactory = LoggerFactory.Create(logging =>
{
    logging.AddOpenTelemetry(options => options.IncludeFormattedMessage = true);

    logging.Services.ConfigureOpenTelemetryLoggerProvider(builder =>
    {
        builder.AddProcessor<MyProcessor>(); // Example of calling a DI-friendly API
        builder.AddGenevaLogExporter(o => o.ConnectionString = "<fill out>");
    });
});
```
